admin/bin/psychometricians-report-v2.js

```diff
@@ -17,15 +19,23 @@ async function main () {
         winston.info('force detected: re-processing all checks')
         await psychometricianReportCacheDataService.sqlDeleteAll()
       }
+      if (process.argv[2] === '-m' || process.argv[3] === '-m') {
+        requiresMarking = true
+      }
     }
 
     winston.info('main: Processing the completed checks')
-
-    // Just process everything in one batch.
-    // TODO: batchify it in a service
-    const checks = await psychometricianReportCacheDataService.sqlFindUnprocessedChecks()
-    if (!checks || !checks.length) return
-    await psychometricianReportService.batchProduceCacheData(checks.map(c => c.id))
+    // Make sure all completed checks are marked and ps-report data cached
+
+    if (requiresMarking) {
+      await completedCheckProcessingService.process()
+    } else {
+      // Just process everything in one batch.
+      // TODO: batchify it in a service
+      const checks = await psychometricianReportCacheDataService.sqlFindUnprocessedChecks()
+      if (!checks || !checks.length) return
```

Comment:
@jon-shipley could replace this with Array.isArray?

Reply:
isArray() would return true for an empty array though